src/blas/backends/syclblas/CMakeLists.txt

```diff
@@ -0,0 +1,197 @@
+  else()
+    message(FATAL_ERROR "Unsupported SYCLBLAS_TUNING_TARGET: '${SYCLBLAS_TUNING_TARGET}'")
+  endif()
+elseif(NUM_TARGETS EQUAL 0)
+  # Enable SYCL-BLAS backend for all devices types
+  set(ENABLE_SYCLBLAS_BACKEND_INTEL_CPU "ON" CACHE INTERNAL "")
+  set(ENABLE_SYCLBLAS_BACKEND_INTEL_GPU "ON" CACHE INTERNAL "")
+  set(ENABLE_SYCLBLAS_BACKEND_AMD_GPU "ON" CACHE INTERNAL "")
+  set(ENABLE_SYCLBLAS_BACKEND_NVIDIA_GPU "ON" CACHE INTERNAL "")
+else()
+  # Try to automatically detect the SYCLBLAS_TUNING_TARGET
+  foreach(SYCL_TARGET IN LISTS SYCL_TARGETS)
+    if(SYCL_TARGETS MATCHES "^intel_gpu" OR SYCL_TARGETS MATCHES "^spir64_gen")
+      set(ENABLE_SYCLBLAS_BACKEND_INTEL_GPU "ON" CACHE INTERNAL "")
+      set(SYCLBLAS_TUNING_TARGET "INTEL_GPU")
+    elseif(SYCL_TARGETS MATCHES "^spir64_x86_64")
+      set(ENABLE_SYCLBLAS_BACKEND_INTEL_CPU "ON" CACHE INTERNAL "")
+    elseif(SYCL_TARGETS MATCHES "^spir64")
+      set(ENABLE_SYCLBLAS_BACKEND_INTEL_CPU "ON" CACHE INTERNAL "")
+      set(ENABLE_SYCLBLAS_BACKEND_INTEL_GPU "ON" CACHE INTERNAL "")
+      set(SYCLBLAS_TUNING_TARGET "INTEL_GPU")
+    elseif(SYCL_TARGETS MATCHES "^amd_gpu" OR SYCL_TARGETS MATCHES "-amd-")
+      set(ENABLE_SYCLBLAS_BACKEND_AMD_GPU "ON" CACHE INTERNAL "")
+      set(SYCLBLAS_TUNING_TARGET "AMD_GPU")
+    elseif(SYCL_TARGETS MATCHES "^nvidia_gpu" OR SYCL_TARGETS MATCHES "-nvidia-")
+      set(ENABLE_SYCLBLAS_BACKEND_NVIDIA_GPU "ON" CACHE INTERNAL "")
+      set(SYCLBLAS_TUNING_TARGET "NVIDIA_GPU")
+    endif()
+  endforeach()
+  # Currently SYCL-BLAS can only be tuned for one type of device.
+  if(NUM_TARGETS GREATER 1)
+    set(SYCLBLAS_TUNING_TARGET "")
+  endif()
+endif()
+
+if(SYCLBLAS_TUNING_TARGET STREQUAL "INTEL_GPU")
+  message(STATUS "Tuning SYCL-BLAS for Intel GPU devices")
+elseif(SYCLBLAS_TUNING_TARGET STREQUAL "AMD_GPU")
+  message(STATUS "Tuning SYCL-BLAS for AMD GPU devices")
+elseif(SYCLBLAS_TUNING_TARGET STREQUAL "NVIDIA_GPU")
+  message(STATUS "Tuning SYCL-BLAS for Nvidia GPU devices")
+else()
+  message(STATUS "SYCL-BLAS is not tuned for any device which can impact performance")
+endif()
+
+# If find_package doesn't work, download SYCL-BLAS from Github. This is
+# intended to make OneMKL easier to use.
+message(STATUS "Looking for sycl_blas")
+find_package(sycl_blas QUIET)
+if (NOT sycl_blas_FOUND)
+  message(STATUS "Looking for sycl_blas - could not find sycl_blas with sycl_blas_DIR")
+  include(FetchContent)
+  set(INSTALL_HEADER_ONLY ON)
+  set(BLAS_BUILD_SAMPLES OFF)
+  set(BLAS_ENABLE_BENCHMARK OFF)
+  set(BLAS_ENABLE_TESTING OFF)
+  set(ENABLE_EXPRESSION_TESTS OFF)
+  # Set the policy to forward variables to SYCL-BLAS configure step
+  set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+  set(FETCHCONTENT_BASE_DIR "${CMAKE_BINARY_DIR}/deps")
+  FetchContent_Declare(
+    sycl-blas
+    GIT_REPOSITORY https://github.com/codeplaysoftware/sycl-blas
+    GIT_TAG        20bfe691be07c9a2a316715094a2441f30e497d7
+  )
+  FetchContent_MakeAvailable(sycl-blas)
```

Comment:
Since it's not part of the project (at least for now). I think downloaded files should be ignored by git so users won't accidently commit them.

Reply:
`FetchContent` pulls the content in the build folder so this is already ignored. This is in the `deps` sub-folder, just like `googletest`.